src/endpoint/s3/s3_utils.js

```diff
@@ -131,7 +131,22 @@ function get_request_xattr(req) {
 }
 
 function set_response_xattr(res, xattr) {
-    _.each(xattr, (val, key) => {
+    let size_for_md_left = config.S3_MD_SIZE_LIMIT;
+    const sliced_xattr = {};
+    const sorted_xattr = Object.keys(xattr).sort().reduce((obj, key) => {
```

Comment:
The sort order is lost once you add keys to object.
I suggest you just keep a sorted list of the keys, and when you need the value to just get it from xattr[key].
```js
const sorted_keys = Object.keys(xattr).sort();
```